src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -528,17 +531,21 @@ private static void cloneImpl(String connectionURL, String scmTag, File checkout
             throw new PluginSourcesUnavailableException("git init was interrupted", e);
         }
 
-        // git fetch ${CONNECTION_URL} ${SCM_TAG}
-        String gitUrl;
-        if (StringUtils.startsWith(connectionURL, "scm:git:")) {
-            gitUrl = StringUtils.substringAfter(connectionURL, "scm:git:");
-        } else {
-            gitUrl = connectionURL;
-        }
+        // until github.com supports fetching abitrary commit-shas we need to fetch anything that
+        // may have this commit
+        // so look at tags, branches and PRs (the latter for incrementals)
+        // git fetch ${CONNECTION_URL} refs/tags/*:refs/origin/tags/*
+        // refs/heads/*:refs/origin/heads/* refs/pull/*/heads:refs/origin/pull/*/heads
         p =
                 new ProcessBuilder()
                         .directory(checkoutDirectory)
-                        .command("git", "fetch", gitUrl, scmTag)
+                        .command(
+                                "git",
+                                "fetch",
+                                gitUrl,
+                                "refs/tags/*:refs/origin/tags/*", // actual releases
```

Comment:
everything is origin.  it doesn't really matter, but when manually testing running git commands in isolation I got a conflict where `master` was not `master`  (should not happen - but we can be safe).